### ISSUE TITLE
Middleware doesn't work with Broadcasting?

### ISSUE THREAD

**Issue by egyptik** [dingo/api#1410]
Trying to setup Broadcasting Authorization, but when I add the auth middleware as such:

```php
Broadcast::routes([ 'middleware' => [ 'api.auth' ] ]);
```

It's throwing the following exception:

`Call to a member function getAuthenticationProviders() on a non-object`

Tried a few things, but can't get this to work. Any ideas?

**Comment by laravuel** [comment]
up

**Comment by MovingGifts** [comment]
@egyptik Did you every figure this out?

**Comment by egyptik** [comment]
Nope sorry, I just stopped using Dingo altogether.

**Comment by MovingGifts** [comment]
Oh okay, thanks for the reply :)

**Comment by MovingGifts** [comment]
@thilanga Any idea how to get `Broadcast` routes to work with dingo's auth middleware?

**Comment by MovingGifts** [comment]
No worries, had to stop using Dingo as well.

**Comment by besingamkb** [comment]
any update on this? got the same issue.

**Comment by MovingGifts** [comment]
@besingamkb I wasn't able to get it to work with Dingo, and since I did not get a response here and there was no way given on how to do this successfully with Dingo, I had to drop Dingo completely and go with laravel's API Resources instead. If you don't get an answer, you can take that route too.
Note: Dingo came out way back in the day (before laravel had it's own API Resources added to its core library, so going forward I think API Resources are the way to go if using laravel)

I hope this helps you and others 👍

**Comment by specialtactics** [comment]
If you can provide some code that we can run to reproduce this problem, and maybe a full stack trace, it will be much more helpful.

**Comment by besingamkb** [comment]
yup I end up creating my own middleware and and authenticate manually there. I think its not the best approach but it works. 
```php
$user = JWTAuth::parseToken()->authenticate();
Auth::login($user);
return $next($request);
```
